lldb/tools/lldb-dap/Handler/StackTraceRequestHandler.cpp

```diff
@@ -178,14 +179,45 @@ void StackTraceRequestHandler::operator()(
   llvm::json::Array stack_frames;
   llvm::json::Object body;
 
+  lldb::SBFormat frame_format = dap.frame_format;
+
+  if (const auto *format = arguments->getObject("format")) {
+    const bool parameters = GetBoolean(format, "parameters").value_or(false);
+    const bool parameter_names =
+        GetBoolean(format, "parameterNames").value_or(false);
+    const bool parameter_values =
+        GetBoolean(format, "parameterValues").value_or(false);
+    const bool line = GetBoolean(format, "line").value_or(false);
+    const bool module = GetBoolean(format, "module").value_or(false);
+    const bool include_all = GetBoolean(format, "includeAll").value_or(false);
```

Comment:
I think this is maybe a different meaning than the spec meant for `includeAll`.

It says:

> Includes all stack frames, including those the debug adapter might otherwise hide.

Which to me, sounds like its about hiding frames that we might not find interesting (e.g. frames from system libraries).

I think if this is `false` you could skip showing the extended backtraces.

Reply:
Great observation, you're right.

Reply:
Its worth noting that today I don't think we skip any frames, however we do use the 'presentationHint' to tweak the frames a little.

* `deemphasize` is used when we don't have the source info for a frame https://github.com/llvm/llvm-project/blob/feaa5aa840dcda69bd4133536142be882f696114/lldb/tools/lldb-dap/JSONUtils.cpp#L774
* `subtle` is used if the frame is artificial or hidden https://github.com/llvm/llvm-project/blob/feaa5aa840dcda69bd4133536142be882f696114/lldb/tools/lldb-dap/JSONUtils.cpp#L795